File: src/1629.py
# https://chat.openai.com/share/c1ae5595-41ba-47a6-839f-b78fe097beae


def partition_list_in_place(lst, x):
    low, mid, high = 0, 0, len(lst) - 1

    while mid <= high:
        if lst[mid] < x:
            lst[low], lst[mid] = lst[mid], lst[low]
            low += 1
            mid += 1
        elif lst[mid] > x:
            lst[mid], lst[high] = lst[high], lst[mid]
            high -= 1
        else:
            mid += 1

    return lst


# Reusing the same example with the in-place method
lst = [9, 12, 3, 5, 14, 10, 10]
x = 10
partitioned_list_in_place = partition_list_in_place(lst, x)
print(partitioned_list_in_place)

File: src/1629.js
// https://chat.openai.com/share/c1ae5595-41ba-47a6-839f-b78fe097beae

function partitionList(list, x) {
  const len = list.length - 1;
  let low = 0;
  let mid = 0;
  let high = len;

  while (mid <= high) {
    if (list[mid] < x) {
      const tmp = list[low];
      list[low] = list[mid];
      list[mid] = tmp;
      low++;
      mid++;
    } else if (list[mid] > x) {
      const tmp = list[high];
      list[high] = list[mid];
      list[mid] = tmp;
      high--;
    } else {
      mid++;
    }
  }

  return list;
}

console.log(partitionList([0, 2, 3, 1], 1));
